integration-tests/src/test/java/org/apache/fineract/integrationtests/DelinquencyBucketsIntegrationTest.java

```diff
@@ -0,0 +1,120 @@
+                responseSpec, jsonRange);
+        final ArrayList<GetDelinquencyRangesResponse> ranges = DelinquencyRangesHelper.getDelinquencyRanges(requestSpec, responseSpec);
+
+        // then
+        assertNotNull(delinquencyRangeResponse01);
+        assertNotNull(ranges);
+        assertEquals(3, ranges.get(0).getMaximumAgeDays(), "Expected Max Age Days to 3");
+    }
+
+    @Test
+    public void testUpdateDelinquencyRanges() {
+        // given
+        String jsonRange = DelinquencyRangesHelper.getAsJSON(1, 3);
+        final PostDelinquencyRangeResponse delinquencyRangeResponse01 = DelinquencyRangesHelper.createDelinquencyRange(requestSpec,
+                responseSpec, jsonRange);
+        jsonRange = DelinquencyRangesHelper.getAsJSON(1, 7);
+
+        // when
+        final PutDelinquencyRangeResponse delinquencyRangeResponse02 = DelinquencyRangesHelper.updateDelinquencyRange(requestSpec,
+                responseSpec, delinquencyRangeResponse01.getResourceId(), jsonRange);
+        final GetDelinquencyRangesResponse range = DelinquencyRangesHelper.getDelinquencyRange(requestSpec, responseSpec,
+                delinquencyRangeResponse01.getResourceId());
+
+        // then
+        assertNotNull(delinquencyRangeResponse01);
+        assertNotNull(delinquencyRangeResponse02);
+        assertNotNull(range);
+        assertNotEquals(3, range.getMaximumAgeDays());
+        assertEquals(7, range.getMaximumAgeDays());
+    }
+
+    @Test
+    public void testDelinquencyBuckets() {
+        // given
+        ArrayList<Integer> rangeIds = new ArrayList<>();
+        String jsonRange = DelinquencyRangesHelper.getAsJSON(1, 3);
+        PostDelinquencyRangeResponse delinquencyRangeResponse = DelinquencyRangesHelper.createDelinquencyRange(requestSpec, responseSpec,
+                jsonRange);
+        rangeIds.add(delinquencyRangeResponse.getResourceId());
+        jsonRange = DelinquencyRangesHelper.getAsJSON(4, 30);
+        delinquencyRangeResponse = DelinquencyRangesHelper.createDelinquencyRange(requestSpec, responseSpec, jsonRange);
+        rangeIds.add(delinquencyRangeResponse.getResourceId());
+        String jsonBucket = DelinquencyBucketsHelper.getAsJSON(rangeIds);
+        PostDelinquencyBucketResponse delinquencyBucketResponse = DelinquencyBucketsHelper.createDelinquencyBucket(requestSpec,
+                responseSpec, jsonBucket);
+
+        // when
+        final ArrayList<GetDelinquencyBucketsResponse> bucketList = DelinquencyBucketsHelper.getDelinquencyBuckets(requestSpec,
+                responseSpec);
+        final GetDelinquencyBucketsResponse delinquencyBucket = DelinquencyBucketsHelper.getDelinquencyBucket(requestSpec, responseSpec,
+                delinquencyBucketResponse.getResourceId());
+
+        // then
+        assertNotNull(bucketList);
+        assertEquals(1, bucketList.size(), "Array size with one element");
+        assertNotNull(delinquencyBucket);
+    }
+}
```

Comment:
@josehernandezfintecheandomx 
Missing test cases:
- Delinquency range is overlapping
- Delinquency range is deleted
- Delinquency range cannot be deleted due it is added to Loan
- Delinquency bucket is already existing
- Delinquency bucket was configured on Loan product
- Delinquency bucket updated
- Delinquency bucket deleted
- Delinquency bucket cannot be deleted due it is configured on Loan product

Reply:
Done! Just pending, the evaluations are there in the code, just more complex

- Delinquency bucket cannot be deleted due it is configured on Loan product
- Delinquency range cannot be deleted due it is added to Loan

Reply:
@josehernandezfintecheandomx I am not sure what you mean here....